internal/table/table_test.go

```diff
@@ -0,0 +1,28 @@
+package table
+
+import (
+	"os"
+	"testing"
+
+	"github.com/ActiveState/cli/internal/output"
+)
+
+func TestStuff(t *testing.T) {
+	out, fail := output.New(string(output.PlainFormatName), &output.Config{
+		OutWriter:   os.Stdout,
+		ErrWriter:   os.Stderr,
+		Colored:     true,
+		Interactive: true,
+	})
+	if fail != nil {
+		t.Fatal(fail)
+	}
+	table := Create([][]string{{"some", "stuff"}})
+	table.AddHeader([]string{"First", "Second"})
+	table.WithHeaderFormat("[HEADING]%s[/RESET]")
```

Comment:
This feels a little awkward to me. The rows and columns are a bit implicit and the headers are added only later on. I'd prefer something a bit more structured like:

```
package main

type Table struct {
	Headers []string
	Rows []Row
	Padding int
}

type Row struct {
	Columns []string
}

func main() {
	_ = Table{
		[]string{"Project", "Owner"},
		[]Row{
			Row{[]string{"cli", "ActiveState"}},
			Row{[]string{"/tmp/cli"}}, // note only one column, we should infer that therefore it would span
		},
		5,
	}
}
```